pkg/network/tracer/tracer_test.go

```diff
@@ -1881,3 +1883,169 @@ func skipIfWindows(t *testing.T) {
 		t.Skip("test unavailable on windows")
 	}
 }
+
+var (
+	disableTLSVerification = sync.Once{}
+)
+
+func writeTempFile(pattern string, content string) (*os.File, error) {
+	f, err := ioutil.TempFile("", pattern)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	if _, err := f.WriteString(content); err != nil {
+		return nil, err
+	}
+
+	return f, nil
+}
+
+func rawConnect(ctx context.Context, t *testing.T, host string, port string) {
+	for {
+		select {
+		case <-ctx.Done():
+			t.Fatalf("failed connecting to port %s:%s", host, port)
+		default:
+			conn, err := net.DialTimeout("tcp", net.JoinHostPort(host, port), time.Second)
+			if err != nil {
+				continue
+			}
+			if conn != nil {
+				conn.Close()
+				return
+			}
+		}
+	}
+
+}
+
+const pythonSSLServerFormat = `import http.server, ssl
+
+class RequestHandler(http.server.BaseHTTPRequestHandler):
+    protocol_version = 'HTTP/1.1'
+    daemon_threads = True
+
+    def do_GET(self):
+        status_code = int(self.path.split("/")[1])
+        self.send_response(status_code)
+        self.end_headers()
+
+server_address = ('127.0.0.1', 8001)
+httpd = http.server.HTTPServer(server_address, RequestHandler)
+httpd.socket = ssl.wrap_socket(httpd.socket,
+                               server_side=True,
+                               certfile='%s',
+                               keyfile='%s',
+                               ssl_version=ssl.PROTOCOL_TLS)
+httpd.serve_forever()
+`
+
+func TestOpenSSLVersions_python_matrix(t *testing.T) {
+	if !httpSupported(t) {
+		t.Skip("HTTPS feature not available on pre 4.14.0 kernels")
+	}
+
+	disableTLSVerification.Do(func() {
+		nethttp.DefaultTransport.(*nethttp.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	})
+	curDir, _ := testutil.CurDir()
+	crtPath := filepath.Join(curDir, "testdata/cert.pem.0")
+	keyPath := filepath.Join(curDir, "testdata/server.key")
+	pythonSSLServer := fmt.Sprintf(pythonSSLServerFormat, crtPath, keyPath)
+	scriptFile, err := writeTempFile("python_openssl_script", pythonSSLServer)
+	require.NoError(t, err)
+	defer scriptFile.Close()
+
+	cmd := exec.Command("python3", scriptFile.Name())
+	go func() {
+		err := cmd.Start()
+		if err != nil {
+			fmt.Println(err)
+		}
+	}()
+	defer func() {
+		if cmd.Process != nil {
+			cmd.Process.Kill()
```

Comment:
What about the test in this case?

Reply:
what?

Reply:
What should happen in the test if this happens? it's an asynchronous operation

Reply:
that's a defer